text/0010-gatsby-docs-localization.md

```diff
@@ -0,0 +1,210 @@
+* Start Date: 2019-09-11
+* RFC PR:
+* Gatsby Issue:
+
+
+## Summary
+
+Begin the process of localizing the documentation of Gatsby into non-English languages.
+
+
+## Motivation
+
+[You Belong Here.](https://www.gatsbyjs.org/blog/2018-09-07-gatsby-values/#you-belong-here)
+
+Gatsby believes that everyone belongs in Open Source, and that includes developers that do not speak English. Having access to translated documentation opens up the ability to create websites to people who don't speak English and don't have the privileges to be able to learn.
+
+
+## Detailed Design
+
+The translations will be done in the [same monorepo](https://github.com/gatsbyjs/gatsby) as the rest of the Gatsby documentation and code. Additional tooling, automation, and documentation will be made to ensure that the languages stay up-to-date and to maintain translation quality.
+
+### Overview of Changes
+
+The following things need to be created:
+
+* [ ] A bot to track changes to the English docs and create pull requests
+* [ ] Script to aid in verifying translation accuracy
+* [ ] Script to set up a new language
+* [ ] Documentation for how to start a new language translation and how to contribute translations
+* [ ] A prioritized list of pages to translate
+
+There are several changes to the repo that need to be made in order to support localization:
+
+* [ ] Move all text strings in `/www` to `/docs` so they can be translated: https://github.com/gatsbyjs/gatsby/issues/17758
+* [ ] Create a `translations/` directory
+* [ ] Make routes such as `/docs` redirect to `/en/docs`
+* [ ] Support routes for translations (such as `/es/docs`)
+* [ ] Support for right-to-left languages
+* [ ] Support for language-specific fonts
+* [ ] UI to toggle between languages
+* [ ] Preserve hashes when switching languages (e.g. `/en/docs/quick-start/#use-the-gatsby-cli` should go to `/es/docs/quick-start/#use-the-gatsby-cli` even though the heading has been translated to Spanish)
+* [ ] Make Algolia only search for results within each language
```

Comment:
I'm not so sure about this. When searching for `starters` even in Spanish, I would expect to show results in English. But if I search for `plantillas`, then I would expect results in Spanish. So I wouldn't restrict the search to only the current language. Specially because maybe not everything is translated. Or maybe you don't know the name of the thing you are looking for in spanish, but find the english equivalent and from there you can discover if there is a translation for another language.

Reply:
I think that's a good point, but if we have ten languages and you search for, say, "themes", you shouldn't get the same article in ten different languages. I think at most you should get the language you're on and the source language (English).

It's definitely something we should research more.